<h1 align = "center" > Up-to-date Heart Attack Analysis And Prediction </h1>

## Project content
1. [Introduction](#1)
    * 1.1 [Examining the Project topics](#2)
    * 1.2 [Recongise varibles in data set](#3)
2. [First Organisation](#4)
   * 2.1 [Require Python labaries](#5)
        * 2.1.1 [Basics labaries](#6)
        * 

### 1. Introduction <a id = 1 ></a>

#### 1.1 Examining the Project topics <a id = 2 ></a>

#### What is a heart attack?
* The medical name of heart attack is "Myocardial infarction".
* Heart attack in short; It is the occlusion of the vessel by plaque-like lesions filled with cholesterol and fat.
* The lesion is called abnormal conditions that occur in the organs where the disease is located.
* As a result of the blockage, the blood flow is completely cut off and a heart attack that can lead to death occurs.

### 1.2 recogizing varibles in the dataset <a id = 3 ></a>

# Heart Attack Dataset - Variable Definitions

## Features

- **Age**: Age of the patient  
- **Sex**: Sex of the patient  
- **exang**: exercise induced angina (1 = yes, 0 = no)  
- **ca**: number of major vessels (0–3)  
### cp (Chest Pain Type)
- 1 = typical angina  
- 2 = atypical angina  
- 3 = non-anginal pain  
- 4 = asymptomatic  
- **trtbps**: resting blood pressure (mm Hg)  
- **chol**: cholesterol (mg/dl)  
- **fbs**: fasting blood sugar > 120 mg/dl (1 = true, 0 = false)  
### rest_ecg
- 0 = normal  
- 1 = ST-T wave abnormality  
- 2 = left ventricular hypertrophy  
- **thalach**: maximum heart rate achieved  
- **target**: 0 = low chance of heart attack, 1 = high chance

# Heart Disease Dataset - Feature Definitions

1. **age** – age in years  

2. **sex** – sex (1 = male, 0 = female)  

3. **cp** – chest pain type  
   - 1 = typical angina  
   - 2 = atypical angina  
   - 3 = non-anginal pain  
   - 0 = asymptomatic  

4. **trestbps** – resting blood pressure (mm Hg on admission to hospital)  

5. **chol** – serum cholesterol (mg/dl)  

6. **fbs** – fasting blood sugar > 120 mg/dl (1 = true, 0 = false)  

7. **restecg** – resting ECG results  
   - 1 = normal  
   - 2 = ST-T wave abnormality  
   - 0 = left ventricular hypertrophy  

8. **thalach** – maximum heart rate achieved  

9. **exang** – exercise induced angina (chest pain) (1 = yes, 0 = no)  

10. **oldpeak** – ST depression induced by exercise relative to rest  

11. **slope** – slope of peak exercise ST segment  
   - 2 = upsloping  
   - 1 = flat  
   - 0 = downsloping  

12. **ca** – number of major vessels (0–3) colored by fluoroscopy  

13. **thal** – thalassemia  
   - 2 = normal  
   - 1 = fixed defect  
   - 3 = reversible defect  

14. **num (target)** – diagnosis of heart disease  
   - 0 = no disease (< 50% narrowing)  
   - 1 = disease (> 50% narrowing)

### First organisation <a id = 4 ></a>

#### 2.1 Require python libaries  <a id = 5 ></a>

#### 2.1.1 Basics Libaries <a id = 6 ></a>

In [1]:
import numpy as np
import pandas as pd
import warnings 
warnings.filterwarnings("ignore")
import matplotlib.pyplot as plt
import seaborn as sns


#### 2.2 Loading the dataset <a id = 7 ></a>

In [2]:
df = pd.read_csv("Heart Attack Data Set.csv")

In [3]:
df

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,63,1,3,145,233,1,0,150,0,2.3,0,0,1,1
1,37,1,2,130,250,0,1,187,0,3.5,0,0,2,1
2,41,0,1,130,204,0,0,172,0,1.4,2,0,2,1
3,56,1,1,120,236,0,1,178,0,0.8,2,0,2,1
4,57,0,0,120,354,0,1,163,1,0.6,2,0,2,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
298,57,0,0,140,241,0,1,123,1,0.2,1,0,3,0
299,45,1,3,110,264,0,1,132,0,1.2,1,0,3,0
300,68,1,0,144,193,1,1,141,0,3.4,1,2,3,0
301,57,1,0,130,131,0,1,115,1,1.2,1,1,3,0


#### 2.3 Inital anaylis on the data set <a id = 8 ></a>

In [4]:
df.head()

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,63,1,3,145,233,1,0,150,0,2.3,0,0,1,1
1,37,1,2,130,250,0,1,187,0,3.5,0,0,2,1
2,41,0,1,130,204,0,0,172,0,1.4,2,0,2,1
3,56,1,1,120,236,0,1,178,0,0.8,2,0,2,1
4,57,0,0,120,354,0,1,163,1,0.6,2,0,2,1


In [5]:
new_columns = ["age", "sex", "cp", "trtbps", "chol", "fbs", "rest_ecg", "thalach", "exang", "oldpeak", "slope", "ca", "thal", "target"]

In [6]:
df.columns = new_columns

In [7]:
df.head()

,age,sex,cp,trtbps,chol,fbs,rest_ecg,thalach,exang,oldpeak,slope,ca,thal,target
0,63,1,3,145,233,1,0,150,0,2.3,0,0,1,1
1,37,1,2,130,250,0,1,187,0,3.5,0,0,2,1
2,41,0,1,130,204,0,0,172,0,1.4,2,0,2,1
3,56,1,1,120,236,0,1,178,0,0.8,2,0,2,1
4,57,0,0,120,354,0,1,163,1,0.6,2,0,2,1


In [8]:
print(" Shape of the Dataset: ", df.shape)

 Shape of the Dataset:  (303, 14)


In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trtbps    303 non-null    int64  
 4   chol      303 non-null    int64  
 5   fbs       303 non-null    int64  
 6   rest_ecg  303 non-null    int64  
 7   thalach   303 non-null    int64  
 8   exang     303 non-null    int64  
 9   oldpeak   303 non-null    float64
 10  slope     303 non-null    int64  
 11  ca        303 non-null    int64  
 12  thal      303 non-null    int64  
 13  target    303 non-null    int64  
dtypes: float64(1), int64(13)
memory usage: 33.3 KB


### 2.3.1 Analysis Outputs(1)

* The Data Set consists of **303 Rows** and **14 Columns**.
* The type of all the variables in the data set are in **numerical format**. (Integer Or Float)
* According to first impressions, there is **no missing value** (NaN Value) in the data set.

# Preperation of EDA

### 3.1 Examining Missing values 


In [11]:
df.isnull().sum()

age         0
sex         0
cp          0
trtbps      0
chol        0
fbs         0
rest_ecg    0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64

In [13]:
is_null_number = []
for i in df.columns: 
    x = df[i].isnull().sum()
    is_null_number.append(x)

pd.DataFrame(is_null_number, index = df.columns, columns=["Total missing value"])

,Total missing value
age,0
sex,0
cp,0
trtbps,0
chol,0
fbs,0
rest_ecg,0
thalach,0
exang,0
oldpeak,0


In [14]:
import missingno

ModuleNotFoundError: No module named 'missingno'